packages/agents/src/agents/Chameleon/index.ts

```diff
@@ -94,6 +93,78 @@ export class ChameleonAgent extends NewAgent<GoalRunArgs> {
         )
     );
   };
+
+  private async shortenLargeMessages(query: string): Promise<void> {
+    for(let i = 2; i < this.context.chat.chatLogs.messages.length ; i++) {
+      const message = this.context.chat.chatLogs.messages[i];
+      if (isLargeMsg(message)) {
+        await this.shortenMessage(message, query);
+      }
+    }
+  }
+
+  private async shortenMessage(message: ChatMessage, query: string): Promise<void> {
+    message.content = await this.advancedFilterText(message.content ?? "", query);
+  }
+
+  // private async filterText(text: string, query: string, context: AgentContext): Promise<string> {
+  //   const chunks = TextChunker.parentDocRetrieval(text, {
+  //     parentChunker: (text: string) => TextChunker.sentences(text),
+  //     childChunker: (parentText: string) =>
+  //       TextChunker.fixedCharacterLength(parentText, { chunkLength: 100, overlap: 15 })
+  //   });
+
+  //   const result = await Rag.standard(chunks, context)
+  //     .selector(x => x.doc)
+  //     .limit(12)
+  //     .sortByIndex()
+  //     .onlyUnique()
+  //     .query(query);
+
+  //   return previewChunks(
+  //     result.map(x => x.metadata.parent),
+  //     2000
+  //   );
+  // }
+
+  private async advancedFilterText(text: string, query: string): Promise<string> {
+    const chunks = TextChunker.parentDocRetrieval(text, {
+      parentChunker: (text: string) => TextChunker.sentences(text),
+      childChunker: (parentText: string) =>
+        TextChunker.fixedCharacterLength(parentText, { chunkLength: 100, overlap: 15 })
+    });
+
+    const result = await Rag.standard(chunks, this.context)
+      .selector(x => x.doc)
+      .limit(48)
+      .sortByIndex()
+      .onlyUnique()
+      .query(query);
+
+    const bigPreview = previewChunks(
+      result.map(x => x.metadata.parent),
+      8000
```

Comment:
NIT: there are a few magic numbers being used. We need to use a number that's based on max context window tokens.